File: python_leaoai/services/Converters/ContactDtoJsonConverter.py
import json
from dataclasses import dataclass, field
from typing import Dict, Any

@dataclass
class ContactDTO:
    Name: str = ""
    Email: str = ""
    Phone: str = ""
    Subject: str = ""
    Message: str = ""
    AdditionalFields: Dict[str, Any] = field(default_factory=dict)

class ContactDtoJsonConverter:

    @staticmethod
    def read(json_str: str) -> ContactDTO:
        """
        Converte JSON string para ContactDTO, incluindo campos adicionais.
        """
        data = json.loads(json_str)
        dto = ContactDTO()

        for key, value in data.items():
            lower_key = key.lower()
            if lower_key == "name":
                dto.Name = value
            elif lower_key == "email":
                dto.Email = value
            elif lower_key == "phone":
                dto.Phone = value
            elif lower_key == "subject":
                dto.Subject = value
            elif lower_key == "message":
                dto.Message = value
            else:
                dto.AdditionalFields[key] = value

        return dto

    @staticmethod
    def write(dto: ContactDTO) -> str:
        """
        Converte ContactDTO em JSON string, incluindo campos adicionais.
        """
        data = {
            "name": dto.Name,
            "email": dto.Email,
            "phone": dto.Phone,
            "subject": dto.Subject,
            "message": dto.Message
        }

        # Adiciona os campos adicionais
        data.update(dto.AdditionalFields)

        return json.dumps(data, ensure_ascii=False)

File: Framework.Services/Converters/ContactDtoJsonConverter.cs
using Framework.Domain.Core.Forms;
using System.Text.Json.Serialization;
using System.Text.Json;

namespace Framework.Services.Converters
{
    public class ContactDtoJsonConverter : JsonConverter<ContactDTO>
    {
        public override ContactDTO Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
        {
            var dto = new ContactDTO();

            // Leitura do objeto JSON
            using (JsonDocument doc = JsonDocument.ParseValue(ref reader))
            {
                foreach (var property in doc.RootElement.EnumerateObject())
                {
                    switch (property.Name.ToLower())
                    {
                        case "name":
                            dto.Name = property.Value.GetString();
                            break;
                        case "email":
                            dto.Email = property.Value.GetString();
                            break;
                        case "phone":
                            dto.Phone = property.Value.GetString();
                            break;
                        case "subject":
                            dto.Subject = property.Value.GetString();
                            break;
                        case "message":
                            dto.Message = property.Value.GetString();
                            break;
                        default:
                            dto.AdditionalFields[property.Name] = property.Value.GetString(); // ou converter para o tipo apropriado
                            break;
                    }
                }
            }

            return dto;
        }

        public override void Write(Utf8JsonWriter writer, ContactDTO value, JsonSerializerOptions options)
        {
            writer.WriteStartObject();

            writer.WriteString("name", value.Name);
            writer.WriteString("email", value.Email);
            writer.WriteString("phone", value.Phone);
            writer.WriteString("subject", value.Subject);
            writer.WriteString("message", value.Message);

            foreach (var field in value.AdditionalFields)
            {
                JsonSerializer.Serialize(writer, field.Value, options);
            }

            writer.WriteEndObject();
        }
    }

}
